src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -667,6 +667,14 @@ default int numCigarElements(){
      */
     byte[] getAttributeAsByteArray( final String attributeName );
 
+    /**
+     * This is used to access the transient attribute store in the underlying SAMRecord.
```

Comment:
Since this is an interface, it should be prescriptive for implementers, and it definitely shouldn't mention `SAMRecord`. The language should match that used in the `set` method below, something like:

```suggestion
     * This is used to access a transient attribute store provided by the underlying implementation. Transient attributes should not be serialized or written out with a record.
```